ibm/service/catalogmanagement/resource_ibm_cm_object.go

```diff
@@ -0,0 +1,607 @@
+)
+
+func ResourceIBMCmObject() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIBMCmObjectCreate,
+		ReadContext:   resourceIBMCmObjectRead,
+		UpdateContext: resourceIBMCmObjectUpdate,
+		DeleteContext: resourceIBMCmObjectDelete,
+		Importer:      &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"catalog_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Catalog identifier.",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The programmatic name of this object.",
+			},
+			"kind": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Kind of object. Options are \"vpe\", \"preset_configuration\", or \"proxy_source\".",
+			},
+			"crn": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The crn for this specific object.",
+			},
+			"url": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The url for this specific object.",
+			},
+			"parent_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The parent for this specific object.",
+			},
+			"label_i18n": &schema.Schema{
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Description: "A map of translated strings, by language code.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"label": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Display name in the requested language.",
+			},
+			"tags": &schema.Schema{
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "List of tags associated with this catalog.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"created": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time this catalog was created.",
+			},
+			"updated": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time this catalog was last updated.",
+			},
+			"short_description": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Short description in the requested language.",
+			},
+			"short_description_i18n": &schema.Schema{
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Description: "A map of translated strings, by language code.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"publish": &schema.Schema{
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "Publish information.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"permit_ibm_public_publish": &schema.Schema{
+							Type:        schema.TypeBool,
```

Comment:
Inner attributes of a computed "publish" can't be decalread as optional they should be computed